RedHat-UX/red-hat-design-system
# fix(card): collapse margins

Closes #1324

## What I did

1. replace padding with margin in card
2. collapse margins for body, header, footer
3. update grid demo


## Testing Instructions

1. confirm that the grid demo cases are comprehensive
2. confirm that header and body *inline* margins are correct
3. confirm that header, body, and footer *block* margins correctly collapse
4. repeat all the above on mobile

## Notes to Reviewers
